packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -0,0 +1,451 @@
+    final double resolvedElevation = resolve<double>((ButtonStyle style) => style?.elevation);
+    final EdgeInsetsGeometry resolvedPadding = resolve<EdgeInsetsGeometry>((ButtonStyle style) => style?.padding);
+    final Size resolvedMinimumSize = resolve<Size>((ButtonStyle style) => style?.minimumSize);
+    final BorderSide resolvedSide = resolve<BorderSide>((ButtonStyle style) => style?.side);
+    final OutlinedBorder resolvedShape = resolve<OutlinedBorder>((ButtonStyle style) => style?.shape);
+
+    // The InkWell's overlayColor parameter is a MaterialStatePropery that resolves against
+    // all of the states that buttons care about except MaterialState.disabled.
+    final MaterialStateProperty<Color> overlayColor = MaterialStateProperty.resolveWith<Color>(
+      (Set<MaterialState> states) {
+        return widgetStyle?.overlayColor?.resolve(states) ?? themeStyle?.overlayColor?.resolve(states) ?? defaultStyle.overlayColor?.resolve(states);
+      }
+    );
+
+    final VisualDensity resolvedVisualDensity = widgetStyle?.visualDensity ?? defaultStyle.visualDensity;
+    final MaterialTapTargetSize resolvedTapTargetSize =  widgetStyle?.tapTargetSize ?? defaultStyle.tapTargetSize;
+
+    final Offset densityAdjustment = resolvedVisualDensity.baseSizeAdjustment;
+    final BoxConstraints effectiveConstraints = resolvedVisualDensity.effectiveConstraints(
+      BoxConstraints(
+        minWidth: resolvedMinimumSize.width,
+        minHeight: resolvedMinimumSize.height,
+      ),
+    );
+    final EdgeInsetsGeometry padding = resolvedPadding.add(
+      EdgeInsets.only(
+        left: densityAdjustment.dx,
+        top: densityAdjustment.dy,
+        right: densityAdjustment.dx,
+        bottom: densityAdjustment.dy,
+      ),
+    ).clamp(EdgeInsets.zero, EdgeInsetsGeometry.infinity);
+
+    final Widget result = ConstrainedBox(
+      constraints: effectiveConstraints,
+      child: Material(
+        elevation: resolvedElevation,
+        textStyle: resolvedTextStyle?.copyWith(color: resolvedForegroundColor),
+        shape: resolvedShape.copyWith(side: resolvedSide),
+        color: resolvedBackgroundColor,
+        shadowColor: resolvedShadowColor,
+        type: resolvedBackgroundColor == null ? MaterialType.transparency : MaterialType.button,
+        animationDuration: widgetStyle?.animationDuration ?? themeStyle?.animationDuration ?? kThemeChangeDuration,
+        clipBehavior: widget.clipBehavior,
+        child: InkWell(
+          onTap: widget.onPressed,
+          onLongPress: widget.onLongPress,
+          onHighlightChanged: _handleHighlightChanged,
+          onHover: _handleHoveredChanged,
+          enableFeedback: widgetStyle?.enableFeedback ?? themeStyle?.enableFeedback ?? true,
+          focusNode: widget.focusNode,
+          canRequestFocus: widget.enabled,
+          onFocusChange: _handleFocusedChanged,
+          autofocus: widget.autofocus,
+          splashFactory: InkRipple.splashFactory,
```

Comment:
Should this be read from the themeData.splashFactory instead?

Reply:
Yes. We're going to need to provide a way for developers to simply configure a theme based on a ColorScheme that uses the 2018 text styles and the InkRipple splashFactory.